service/src/test/java/uk/gov/hmcts/reform/fpl/service/payment/PaymentServiceTest.java

```diff
@@ -300,4 +367,66 @@ void resetInvocations() {
             .service(FPL)
             .siteId("TEST_SITE_ID");
     }
+
+    private CreditAccountPaymentRequest buildCreditAccountPaymentRequest(String caseReference,
+                                                                         String customerReference,
+                                                                         FeeDto feeDto) {
+        return CreditAccountPaymentRequest.builder()
+           .accountNumber("PBA123")
+           .caseReference(defaultIfBlank(caseReference, BLANK_PARAMETER_VALUE))
+           .customerReference(defaultIfBlank(customerReference, BLANK_PARAMETER_VALUE))
+           .ccdCaseNumber(String.valueOf(CASE_ID))
+           .currency(GBP)
+           .description("Payment for case: " + CASE_ID)
+           .organisationName("Example Local Authority")
+           .service(FPL)
+           .siteId("TEST_SITE_ID")
+           .amount(feeDto.getCalculatedAmount())
+           .fees(List.of(feeDto)).build();
+    }
+
+    private CaseData buildCaseData(String clientCode, String customerReference, C2ApplicationType type) {
+        return CaseData.builder()
+            .caseLocalAuthority("LA")
+            .c2DocumentBundle(List.of(element(C2DocumentBundle.builder()
+                .type(type)
+                .pbaNumber("PBA123")
+                .clientCode(clientCode)
+                .fileReference(customerReference)
+                .build())))
+            .build();
+    }
+
+    private CreditAccountPaymentRequest buildCreditAccountPaymentRequestForC110Application(String caseReference,
+                                                                                           String customerReference) {
+        return CreditAccountPaymentRequest.builder()
+            .accountNumber("PBA123")
+            .caseReference(defaultIfBlank(caseReference, BLANK_PARAMETER_VALUE))
```

Comment:
Related to omy above comments, but again Blank parameter value, shouldn't come from the class there is no guarantee it's correct value.